rs/execution_environment/src/execution/update.rs

```diff
@@ -341,26 +359,22 @@ impl UpdateHelper {
 
         validate_message(&canister, &original.method)?;
 
-        if let CanisterCallOrTask::Call(_) = original.call_or_task {
-            // TODO(RUN-957): Enforce the limit in heartbeat and timer after
```

Comment:
There's still one more TODO(RUN-957) in the code to be resolved. CC @dragoljub-duric

Reply:
But in my opinion, it seems safer to not enforce the limit _during_ a system task, i.e., simply drop the other TODO(RUN-957), instead of trapping _during_ a system task. To be clear: I do believe this PR makes sense if we ensure that system tasks do not even start executing if the limit is exceeded (the open conversation on this PR).